### ISSUE TITLE
Cannot use custom class with TypeAhead

### ISSUE THREAD

**Issue by illuminated** [danvick/flutter_form_builder#244]
If I'm providing a List of Strings I have no issue using the TypeAhead but when I use a custom type, I get that 
```
The following assertion was thrown building _FormScope:
type 'String' is not a subtype of type 'StaticCountry'
```
This is my TypeAhead widget:
```
                      FormBuilderTypeAhead(
                        decoration: InputDecoration(
                          labelText: "Country",
                        ),
                        attribute: 'country',
                        selectionToTextTransformer: (StaticCountry country) => country.name,
                        itemBuilder: (context, StaticCountry country) {
                          return ListTile(
                            title: Text(country.name),
                          );
                        },
                        suggestionsCallback: (query) async {
                          if (query.isNotEmpty) {
                            return await model.findStaticCountries(query);
                          } else {
                            return await model.loadStaticCountries();
                          }
                        },
                      ),
```
The loadStaticCountries and findStaticCountries are returning a Future<List<StaticCountry>>:
```
  Future<List<StaticCountry>> loadStaticCountries() async {
    String jsonString = await _loadStaticCountryAsset();
    final jsonResponse = json.decode(jsonString) as List;
    return jsonResponse
        .map((staticJsonCountry) => StaticCountry.fromJson(staticJsonCountry))
        .toList(growable: false);
  }
  Future<List<StaticCountry>> findStaticCountries(String pattern) async {
    List<StaticCountry> allCountries = await loadStaticCountries();
    return allCountries.where((aCountry) {
      return aCountry.name.contains(pattern);
    }).toList(growable: false);
  }
```
And my StaticCountry class is nothing special:
```
@JsonSerializable(explicitToJson: true)
class StaticCountry {
  final String name;
  final String alpha2Code;
  final String alpha3Code;
  final List<String> topLevelDomain;
  final String capital;
  final String region;
  final String subregion;
  final int population;
  final String flag;
  final List<String> callingCodes;
  final String nativeName;

  StaticCountry(
    this.name,
    this.alpha2Code,
    this.alpha3Code,
    this.topLevelDomain,
    this.capital,
    this.region,
    this.subregion,
    this.population,
    this.flag,
    this.callingCodes,
    this.nativeName,
  );

  factory StaticCountry.fromJson(Map<String, dynamic> json) =>
      _$StaticCountryFromJson(json);

  Map<String, dynamic> toJson() => _$StaticCountryToJson(this);
}
```

I'm loading the list of countries from a static json file as an asset, that's where the StaticCountry name comes from. 

Anyways, I've lost now two days on trying to make this work. I think I've tried everything I could think of.

Any help appreciated.

**Comment by brizaldi** [comment]
this is just a workaround but i solved it by adding `initialData`, something like this:

`initialData: StaticCountry(),`

**Comment by awhitford** [comment]
I just submitted PR #320 to fix this properly.  This is the same issue I discovered in #319.

**Comment by awhitford** [comment]
Closing because fix is part of **3.11.1**.
